Files java-config-1.3.0.vanilla/java_config/jc_exceptions.pyc and java-config-1.3.0/java_config/jc_exceptions.pyc differ
Files java-config-1.3.0.vanilla/java_config/jc_output.pyc and java-config-1.3.0/java_config/jc_output.pyc differ
diff -Nru java-config-1.3.0.vanilla/java_config/jc_util.py java-config-1.3.0/java_config/jc_util.py
--- java-config-1.3.0.vanilla/java_config/jc_util.py	2006-08-03 01:32:30.000000000 +0200
+++ java-config-1.3.0/java_config/jc_util.py	2006-08-03 01:50:33.000000000 +0200
@@ -66,8 +66,31 @@
 get_jvm_env_var.user_env_dir = os.path.join(env.get('HOME', 'THERE_IS_NO_HOME'), '.gentoo')
 
 def get_java_home():
+	for f in os.listdir('/etc/env.d/java/'):
+		vmHandle = ''
+		try:
+			fh = open('/etc/env.d/java/' + f, 'r')
+			for line in fh.readlines():
+				line = line.strip()
+				if line[0:8] == 'VMHANDLE':
+					vmHandle = line[10:-1]
+					break
+			fh.close()
+		except:
+			continue
+		if vmHandle == os.environ['VMHANDLE']:
+			try:
+				fh = open('/etc/env.d/java/' + f, 'r')
+				for line in fh.readlines():
+					line = line.strip()
+					if line[0:9] == 'JAVA_HOME':
+						return line[10:]
+						break
+				fh.close()
+			except:
+				continue
 
-	return get_jvm_env_var ('JAVA_HOME')
+	return None
 
 
 JAVA_HOME = get_java_home()
